packages/shared/src/components/plus/GiftReceivedPlusModal.tsx

```diff
@@ -0,0 +1,129 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+import { useQuery } from '@tanstack/react-query';
+import type { ModalProps } from '../modals/common/Modal';
+import { Modal } from '../modals/common/Modal';
+
+import type { UserShortProfile } from '../../lib/user';
+import { PlusTitle } from './PlusTitle';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../typography/Typography';
+import CloseButton from '../CloseButton';
+import { ButtonSize, ButtonVariant } from '../buttons/common';
+import { cloudinaryGiftedPlusModalImage } from '../../lib/image';
+import { PlusList } from './PlusList';
+import { Button } from '../buttons/Button';
+import { getPlusGifterUser } from '../../graphql/users';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { ProfileImageSize, ProfilePicture } from '../ProfilePicture';
+import Link from '../utilities/Link';
+
+interface GiftPlusModalProps extends ModalProps {
+  user?: UserShortProfile;
+}
+
+const GifterProfile = ({ gifter }: { gifter: UserShortProfile }) => (
+  <Link href={`/${gifter.username}`} passHref>
+    <a
+      className="flex items-center gap-2"
+      title={`View ${gifter.username}'s profile`}
+    >
+      <ProfilePicture user={gifter} size={ProfileImageSize.Medium} />
+      <Typography
+        bold
+        color={TypographyColor.Primary}
+        type={TypographyType.Callout}
+      >
+        {gifter.name}
+      </Typography>
+      <Typography
+        type={TypographyType.Footnote}
+        color={TypographyColor.Secondary}
+      >
+        @{gifter.username}
+      </Typography>
+    </a>
+  </Link>
+);
+
+const OpenSquadButton = () => (
+  <Button
+    className="w-full"
+    href="/squads/plus"
+    tag="a"
+    variant={ButtonVariant.Primary}
+  >{`See what's inside`}</Button>
+);
+
+export function GiftReceivedPlusModal({
+  user,
+  onRequestClose,
+  ...props
+}: GiftPlusModalProps): ReactElement {
+  const { data: gifter, isLoading } = useQuery({
+    queryKey: generateQueryKey(RequestKey.GifterUser),
+    queryFn: getPlusGifterUser,
+    enabled: Boolean(user.isPlus),
+  });
+
+  if (!gifter || isLoading) {
+    return null;
+  }
```

Comment:
Maybe add a modal like loader so user knows something is opening, see https://github.com/dailydotdev/apps/pull/4128/files#diff-ea1df30479911bb7e3f5fcb0727c06eac149de4aa7f08e1dae88b4ce5c217a9cR297-R303

Reply:
non blocking